src/test/java/CalendarServletTest.java

```diff
@@ -109,47 +106,39 @@ public void setUp() throws IOException, GeneralSecurityException {
   public void noCalendarEvent() throws IOException, ServletException {
     // Test case where there are no events in the user's calendar
     Mockito.when(calendarClient.getCalendarEvents()).thenReturn(NO_EVENT);
-    servlet.doGet(request, response);
-    printWriter.flush();
-    String actualString = stringWriter.toString();
-    Type type = new TypeToken<List<Event>>() {}.getType();
-    List<Event> actual = gson.fromJson(actualString, type);
-    Assert.assertEquals(NO_EVENT, actual);
+    checkReturn(NO_EVENT);
   }
 
   @Test
   public void firstTwoEvents() throws IOException, ServletException {
     // Test case where there are two events with defined summaries
+    // Events must be returned in order of retrieval - JSON includes tasks in desired order
     Mockito.when(calendarClient.getCalendarEvents()).thenReturn(EVENT_ONE_TWO);
-    servlet.doGet(request, response);
-    printWriter.flush();
-    String actualString = stringWriter.toString();
-    Type type = new TypeToken<List<Event>>() {}.getType();
-    List<Event> actual = gson.fromJson(actualString, type);
-    Assert.assertEquals(EVENT_ONE_TWO, actual);
+    checkReturn(EVENT_ONE_TWO);
   }
 
   @Test
   public void undefinedEvent() throws IOException, ServletException {
     // Test case where there is an event with no summary
     Mockito.when(calendarClient.getCalendarEvents()).thenReturn(EVENT_UNDEFINED);
-    servlet.doGet(request, response);
-    printWriter.flush();
-    String actualString = stringWriter.toString();
-    Type type = new TypeToken<List<Event>>() {}.getType();
-    List<Event> actual = gson.fromJson(actualString, type);
-    Assert.assertEquals(EVENT_UNDEFINED, actual);
+    checkReturn(EVENT_UNDEFINED);
   }
 
   @Test
   public void allEvent() throws IOException, ServletException {
     // Test case where there are two defined and an undefined event
+    // Events must be returned in order of retrieval - JSON includes tasks in desired order
     Mockito.when(calendarClient.getCalendarEvents()).thenReturn(EVENT_ALL);
+    checkReturn(EVENT_ALL);
+  }
+
+  public void checkReturn(List<Event> EXPECTED_EVENT) throws IOException, ServletException {
+    // Method that handles the once the Calendar Client has been mocked
     servlet.doGet(request, response);
     printWriter.flush();
     String actualString = stringWriter.toString();
     Type type = new TypeToken<List<Event>>() {}.getType();
     List<Event> actual = gson.fromJson(actualString, type);
-    Assert.assertEquals(EVENT_ALL, actual);
+    Assert.assertEquals(EXPECTED_EVENT, actual);
```

Comment:
Super nit: I'd rather leave the assertion outside the helper method so it is easier to debug. Knowing that the assertion inside the helper method failed isn't enough to know which exact test failed (you would still need to check the stack trace). Having the assertions inside the tests remove that need.